### PR TITLE
add a capabilities module

### PR DESCRIPTION
**Pull request by yelhamer**
This PR moves the capability extraction logic from \`main.py\` to a new "capabilities" module.

### CODE REVIEW

**Comment by williballenthin** [comment]
I like the idea of splitting out this logic into its own module, because:

  1. main.py is way too big, and,
  2. this logic is re-used by other scripts/tools, so it makes sense to provide a stable module path

I'd propose that this new module be `capa.capabilities` (not `capa.features.capabilities`). Thoughts @mr-tz @mike-hunhoff @yelhamer?

@yelhamer please also update the scripts that use these routines:

```
scripts/profile-time.py
117:            capa.main.find_capabilities(rules, extractor, disable_progress=True)

scripts/capa_as_library.py
35:def find_subrule_matches(doc: rd.ResultDocument) -> Set[str]:
178:    capabilities, counts = capa.main.find_capabilities(rules, extractor, disable_progress=True)

scripts/lint.py
369:    capabilities, _ = capa.main.find_capabilities(ctx.rules, extractor, disable_progress=True)

scripts/show-capabilities-by-function.py
189:    capabilities, counts = capa.main.find_capabilities(rules, extractor)

scripts/bulk-process.py
139:    capabilities, counts = capa.main.find_capabilities(rules, extractor, disable_progress=True)
```

**Comment by williballenthin** [comment]
Furthermore, does it make sense to break out any other logic from main? Here's whats left:

```
timing
set_vivisect_log_level
has_rule_with_namespace
is_internal_rule
is_file_limitation_rule
has_file_limitation
is_supported_format
is_supported_arch
get_arch
is_supported_os
get_os
get_meta_str
is_running_standalone
get_default_root
get_default_signatures
get_workspace
get_extractor
get_file_extractors
is_nursery_rule_path
collect_rule_file_paths
on_load_rule_default
get_rules
get_signatures
get_sample_analysis
collect_metadata
compute_dynamic_layout
compute_static_layout
compute_layout
install_common_args
handle_common_args
main
ida_main
ghidra_main
```

**Comment by mr-tz** [comment]
> I'd propose that this new module be capa.capabilities (not capa.features.capabilities). Thoughts @mr-tz @mike-hunhoff @yelhamer?

agreed

I think we should use this as a start to a larger refactor significantly reducing the size of main and making capa much easier to use as a library (and within our own code), see for example #1813 as well as many code duplication in our scripts.
Let's wait on that big refactor till all the current big tasks and PRs have been addressed though.

File: capa/capabilities/common.py
Comment (williballenthin) [inline capa/capabilities/common.py:19]:
```suggestion
logger = logging.getLogger(__name__)
```

Comment (williballenthin) [reply capa/capabilities/common.py:19]: so that the logging prefix looks like `capa.capabilities.common` which is nice for debugging

Comment (williballenthin) [reply capa/capabilities/common.py:19]: and, we don't have to type this name ourselves, so it can be easily copy-pasted-moved to other files.

File: capa/capabilities/static.py
Comment (williballenthin) [inline capa/capabilities/static.py:26]:
```suggestion
logger = logging.getLogger(__name__)
```

File: capa/rules/__init__.py
Comment (williballenthin) [inline capa/rules/__init__.py:947]: ah, i guess we had this in `main` because the capa-rules layout concepts aren't otherwise referenced in `rules.py`, so it sort of breaks the abstraction. but, i think its better to keep related code together, rather than have this be totally perfect.

**Review by williballenthin — APPROVED** [review]
looks great, thanks @yelhamer!

File: capa/rules/__init__.py
Comment (yelhamer) [reply capa/rules/__init__.py:947]: ah I see, maybe we should plan to add a "helpers.py" file in capa.rules with such functions in the future?

**Comment by williballenthin** [comment]
@mr-tz i'd like to have your LGTM here too before we merge

**Comment by mr-tz** [comment]
reviewing now!

File: capa/rules/__init__.py
Comment (mr-tz) [inline capa/rules/__init__.py:943]: this function seems to be unused right now

File: tests/test_main.py
Comment (mr-tz) [inline tests/test_main.py:404]: please add this as well

File: capa/main.py
Comment (mr-tz) [inline capa/main.py:137]: move this similar to `is_internal_rule` / `is_file_limitation_rule`?

**Review by mr-tz — CHANGES_REQUESTED** [review]
great changes, just a few minor issues I've noticed

File: capa/rules/__init__.py
Comment (yelhamer) [reply capa/rules/__init__.py:943]: yeah, it was included in main.py as well. should I get rid of it?

File: capa/main.py
Comment (yelhamer) [reply capa/main.py]: i believe this function is also unused. I've moved it to `capa.rules.RuleSet`, should I get rid of it?

Comment (mr-tz) [reply capa/main.py]: yes, if both are unused I think we should remove them

**Review by mr-tz — APPROVED** [review]
LGTM!

**Pull request merged**
